susemanager-utils/susemanager-sls/src/modules/mgr_caasp_manager.py

```diff
@@ -0,0 +1,347 @@
+# -*- coding: utf-8 -*-
+'''
+SUSE Manager CaaSP Cluster Manager module for Salt
+
+'''
+from __future__ import absolute_import
+
+
+import logging
+import os
+import subprocess
+
+import salt.utils.stringutils
+import salt.utils.timed_subprocess
+
+try:
+    from salt.utils.path import which
+except ImportError:
+    from salt.utils import which
+
+from salt.utils.dictupdate import merge_list
+from salt.exceptions import CommandExecutionError
+
+
+log = logging.getLogger(__name__)
+
+__virtualname__ = 'caasp'
+
+DEFAULT_TIMEOUT = 1200
+
+
+def __virtual__():
+    '''
+    This module is always enabled while 'skuba' CLI tools is available.
+    '''
+    return __virtualname__ if which('skuba') else (False, 'skuba is not available')
+
+
+def _call_skuba(skuba_cluster_path,
+                cmd_args,
+                timeout=DEFAULT_TIMEOUT,
+                **kwargs):
+
+    log.debug("Calling Skuba CLI: 'skuba {}' - Timeout: {}".format(cmd_args, timeout))
+    try:
+        skuba_proc = salt.utils.timed_subprocess.TimedProc(
+            ["skuba"] + cmd_args.split(),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            timeout=timeout,
+            cwd=skuba_cluster_path,
+        )
+        skuba_proc.run()
+        return skuba_proc
+    except Exception as exc:
+        error_msg = "Unexpected error while calling skuba: {}".format(exc)
+        log.error(error_msg)
+        raise CommandExecutionError(error_msg)
+
+
+def _sanitize_skuba_output_values(items):
+    ret = []
+    for i in items:
+        if i.lower() == 'no':
+            ret.append(False)
+        elif i.lower() == 'yes':
+            ret.append(True)
+        elif i.lower() == '<none>':
+            ret.append(None)
+        else:
+            ret.append(i)
+    return ret
+
+
+def list_nodes(skuba_cluster_path,
+               timeout=DEFAULT_TIMEOUT,
+               **kwargs):
+    skuba_proc = _call_skuba(skuba_cluster_path, "cluster status")
```

Comment:
I think timeout  `timeout` should be passed as arg to every call to `_call_skuba`, am I assuming correctly?

Reply:
You're correct. Good catch! I've missed that :)